app/post_results.go

```diff
@@ -259,13 +255,85 @@ func extractAndVerifyCertForPayload(ctx context.Context,
 	}
 
 	cert := certs[0]
+
 	// Verify that cert isn't expired.
-	if err = cosign.CheckExpiry(cert, time.Unix(*entry.IntegratedTime, 0)); err != nil {
-		return nil, fmt.Errorf("error during cosign.CheckExpiry: %w", err)
+	integratedTime := time.Unix(entry.IntegratedTime, 0)
+	if cert.NotAfter.Before(integratedTime) {
+		return nil, fmt.Errorf("certificate expired before signatures were entered in log: %s is before %s",
+			cert.NotAfter, integratedTime)
+	}
+	if cert.NotBefore.After(integratedTime) {
+		return nil, fmt.Errorf("certificate was issued after signatures were entered in log: %s is after %s",
+			cert.NotAfter, integratedTime)
 	}
 	return cert, nil
 }
 
+func getUUIDsByPayload(ctx context.Context, payload []byte) ([]string, error) {
+	payloadSHA := sha256.Sum256(payload)
+	rekorPayload := struct {
+		Hash string `json:"hash"`
+	}{
+		Hash: fmt.Sprintf("sha256:%s", hex.EncodeToString(payloadSHA[:])),
+	}
+	jsonPayload, err := json.Marshal(rekorPayload)
+	if err != nil {
+		return nil, fmt.Errorf("marshaling json payload: %w", err)
+	}
+
+	rekorReq, err := http.NewRequestWithContext(ctx,
+		http.MethodPost,
+		"https://rekor.sigstore.dev/api/v1/index/retrieve",
+		bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return nil, fmt.Errorf("creating new HTTP request: %w", err)
+	}
+	rekorReq.Header.Add("Content-Type", "application/json")
+	rekorReq.Header.Add("accept", "application/json")
+	resp, err := http.DefaultClient.Do(rekorReq)
+	if err != nil {
+		return nil, fmt.Errorf("looking up Rekor index: %w", err)
+	}
+	defer resp.Body.Close()
+
+	var rekorResult []string
+	if err := json.NewDecoder(resp.Body).Decode(&rekorResult); err != nil {
+		return nil, fmt.Errorf("decoding Rekor response: %w", err)
+	}
+
+	return rekorResult, nil
+}
+
+func getTLogEntry(ctx context.Context, uuid string) (*tlogEntry, error) {
+	rekorReq, err := http.NewRequestWithContext(ctx,
+		http.MethodGet,
+		fmt.Sprintf("https://rekor.sigstore.dev/api/v1/log/entries/%s", uuid),
```

Comment:
The same comment as above. We should probably use the `staging`

Reply:
I disagree, staging can get wiped and may be unstable; `rekor.sigstore.dev` is meant to be the production endpoint (albeit we are still operating in a beta capacity).

Reply:
OK, Thanks